packages/run-protocol/src/vpool-xyk-amm/pool.js

```diff
@@ -1,301 +1,328 @@
+
+    state.liqTokenSupply -= liquidityValueIn;
+
+    poolSeat.incrementBy(
+      userSeat.decrementBy(harden({ Liquidity: liquidityIn })),
+    );
+    userSeat.incrementBy(
+      poolSeat.decrementBy(
+        harden({
+          Central: centralTokenAmountOut,
+          Secondary: tokenKeywordAmountOut,
+        }),
+      ),
+    );
+    state.zcf.reallocate(userSeat, poolSeat);
+
+    userSeat.exit();
+    updateUpdaterState(state.updater, facets.pool);
+    return 'Liquidity successfully removed.';
+  },
+  getNotifier: ({ state: { notifier } }) => notifier,
+  updateState: ({ state: { updater }, facets: { pool } }) => {
+    return updater.updateState(pool);
+  },
+  getToCentralPriceAuthority: ({ state }) => state.toCentralPriceAuthority,
+  getFromCentralPriceAuthority: ({ state }) => state.fromCentralPriceAuthority,
+  getVPool: ({ facets }) => facets.singlePool,
+};
+
+/** @param {MethodContext} context */
+const finish = context => {
+  const { notifier, secondaryBrand, centralBrand, timer, zcf } = context.state;
+  const quoteIssuerKit = context.state.quoteIssuerKit;
+
+  const getInputPriceForPA = (amountIn, brandOut) => {
+    return publicPrices(
+      context.facets.singlePool.getPriceForInput(
+        amountIn,
+        // @ts-ignore confused about whether it needs a context
+        AmountMath.makeEmpty(brandOut),
+      ),
+    );
+  };
+  const getOutputPriceForPA = (brandIn, amountout) =>
+    publicPrices(
+      context.facets.singlePool.getPriceForOutput(
+        AmountMath.makeEmpty(brandIn),
+        // @ts-ignore confused about whether it needs a context
+        amountout,
+      ),
+    );
+
+  const toCentralPriceAuthority = makePriceAuthority(
+    getInputPriceForPA,
+    getOutputPriceForPA,
+    secondaryBrand,
+    centralBrand,
+    timer,
+    zcf,
+    notifier,
+    quoteIssuerKit,
+  );
+  const fromCentralPriceAuthority = makePriceAuthority(
+    getInputPriceForPA,
+    getOutputPriceForPA,
+    centralBrand,
+    secondaryBrand,
+    timer,
+    zcf,
+    notifier,
+    quoteIssuerKit,
+  );
+
+  // @ts-ignore declared read-only, set value once
+  context.state.toCentralPriceAuthority = toCentralPriceAuthority;
+  // @ts-ignore declared read-only, set value once
+  context.state.fromCentralPriceAuthority = fromCentralPriceAuthority;
+};
+
 /**
  * @param {ZCF} zcf
- * @param {(brand: Brand) => boolean} isInSecondaries
- * @param {(brand: Brand, pool: XYKPool) => void} initPool
  * @param {Brand} centralBrand
  * @param {ERef<Timer>} timer
  * @param {IssuerKit} quoteIssuerKit
- * @param {() => bigint} getProtocolFeeBP - retrieve governed protocol fee value
- * @param {() => bigint} getPoolFeeBP - retrieve governed pool fee value
- * @param {ZCFSeat} protocolSeat
+ * @param {import('./multipoolMarketMaker.js').AMMParamGetters} paramAccessor retrieve governed params
+ * @param {ZCFSeat} protocolSeat seat that holds collected fees
  */
-export const makeAddPool = (
+export const makePoolMaker = (
```

Comment:
"make … maker" is a smell to me. Fortunately we have a more specific thing we're making: a "kind" is an instance of the thing a `maker` returned by `defineKind` makes. Cutting to the chase:
```suggestion
export const definePoolKind = (
```

And then in `addPool.js`,
```
  const makePool = definePoolKind(
```

Reply:
I'm used to seeing `makeFooMaker`. I've seen `FooFactoryFactory`.  can you summarize what bad practice it indicates?

Reply:
Cognitive burden distinguishing the two things named same (maker, factory). But that's a weak argument.

The better reason to use `definePoolKind` is because it's returning what a `defineKind` returns.

Reply:
> The better reason to use definePoolKind is because it's returning what a defineKind returns.
I don't disagree that `definePoolKind` is better here. I was looking for a reason to be suspicious of `makeFooMaker` in general.

Reply:
> I was looking for a reason to be suspicious of makeFooMaker in general.

My error in casting suspicion on it by using the term "smell". I didn't mean a "code smell" (i.e. "a surface indication that usually corresponds to a deeper problem in the system"). I just meant "stinky". :)